OFBIZ-13212-framework-update

arashpreet0011 opened the pull request:
Implemented: Modified JWTManager.java to generate and validate refresh token (OFBIZ-13212)

Explanation: This implementation will allow user to get refresh token while login.

https://github.com/apache/ofbiz-plugins/pull/135 is related

arashinsonix committed:
OFBIZ-13212-framework-update

- Modified framework/webapp/src/main/java/org/apache/ofbiz/webapp/control/JWTManager.java to generate and validate refresh token.

JacquesLeRoux reviewed (changes requested):
Hi @arashpreet0011,

security.jwt.refreshToken.expireTime is missing. Why 86400 as default timeout?

JacquesLeRoux commented:
Don't worry about CodeQL failures. They exist for few weeks, are not related to us, so we ignore them for now and use those generated before.

arashpreet0011 commented:
> Hi @arashpreet0011,
> 
> security.jwt.refreshToken.expireTime is missing. Why 86400 as default timeout?

While generating an access token, the default validity period is set to 1800 seconds (30 minutes). I have configured the refresh token validity to 86,400 seconds (24 hours) following the same approach, ensuring consistency in token expiration settings.

JacquesLeRoux commented:
You need to add security.jwt.refreshToken.expireTime in security.properties.
 I guess it's on your side, but not here ;)

arashinsonix committed:
OFBIZ-13212-framework-update

- Added security.jwt.refresh.token.expireTime property in security for refresh token.

arashpreet0011 commented:
> You need to add security.jwt.refreshToken.expireTime in security.properties. I guess it's on your side, but not here ;)

I added property security.jwt.refresh.token.expireTime in security.properties file. 

But expire time for createJwt method which is ofbiz default is also passed int the method as below - 

`Integer.parseInt(EntityUtilProperties.getPropertyValue("security", "security.jwt.token.expireTime", "1800", delegator));`

In same way I added this property as below - 

```
int refreshTokenExpireTime = Integer.parseInt(EntityUtilProperties.getPropertyValue("security",
                "security.jwt.refresh.token.expireTime", "86400", delegator));
```
                
Below is latest commit - 

https://github.com/apache/ofbiz-framework/commit/4383971767de4833026eece43a1a0b55c99b5cad

JacquesLeRoux commented on framework/webapp/src/main/java/org/apache/ofbiz/webapp/control/JWTManager.java:
```diff
@@ -404,4 +404,18 @@
+
+    public static String createRefreshToken(Delegator delegator, String userLoginId) {
+        int refreshTokenExpireTime = Integer.parseInt(EntityUtilProperties.getPropertyValue("security",
+                "security.jwt.refreshToken.expireTime", "86400", delegator));
```
Already dismissed
